finalfrontier/src/vocab.rs

```diff
@@ -388,6 +392,42 @@ where
         SimpleVocab::new(builder.config, types, builder.n_items)
     }
 }
+/// Constructs a `SimpleVocab<S>` from a `VocabBuilder<T>` where `T: Into<S>`.
+impl<T, S> From<VocabBuilder<SimpleSizedVocabConfig, T>> for SimpleVocab<S>
+where
+    T: Hash + Eq + Into<S>,
+    S: Hash + Eq + Clone + Ord,
+{
+    fn from(builder: VocabBuilder<SimpleSizedVocabConfig, T>) -> Self {
+        let config = builder.config;
+
+        let mut types: Vec<_> = builder
+            .items
+            .into_iter()
+            .map(|(item, count)| CountedType::new(item.into(), count))
+            .collect();
+        types.sort_unstable_by(|w1, w2| w2.cmp(&w1));
+        let vocab_size = config.vocab_size;
+        let mut last_index = vocab_size - 1;
```

Comment:
This can be replaced by something like (totally untested, probably doesn't even compile, but should give the idea):

```rust
if let Some(last_type) = types.get(last_index) {
  let cutoff_point = match types[..last_index].binary_search_by(|other_type| {
    use Ordering::*;
    match other_type.count().cmp(&last_type.count()) {
      Less => Less,
      Equal => Greater,
      Greater => Greater,
    }
  }) {
    Ok(idx) => idx,
    Err(idx) => idx,
  }
}
```

Factored out as a function for reusability.